libfwupdplugin/fu-efi-file.c

```diff
@@ -241,11 +241,13 @@ fu_efi_file_write(FuFirmware *firmware, GError **error)
 	blob = fu_efi_file_write_sections(firmware, error);
 	if (blob == NULL)
 		return NULL;
-	if (!fwupd_guid_from_string(fu_firmware_get_id(firmware),
-				    &guid,
-				    FWUPD_GUID_FLAG_MIXED_ENDIAN,
-				    error))
-		return NULL;
+	if (fu_firmware_get_id(firmware) != NULL) {
```

Comment:
isn't this a programmer error?  Should it be noisier?

Reply:
I think in this case a NULL GUID is fine, but point heeded.